pkg/vpn_client/iptables.go

```diff
@@ -35,6 +35,30 @@ func SetIPTableRules(log logr.Logger, cfg config.VPNClient) error {
 				if err != nil {
 					return err
 				}
+				err = ipTable.AppendUnique("nat", "PREROUTING", "--in-interface", forwardDevice, "-d", constants.ShootPodNetworkMapped, "-j", "NETMAP", "--to", cfg.ShootPodNetwork.String())
+				if err != nil {
+					return err
+				}
+				err = ipTable.AppendUnique("nat", "POSTROUTING", "--out-interface", forwardDevice, "-s", cfg.ShootPodNetwork.String(), "-j", "NETMAP", "--to", constants.ShootPodNetworkMapped)
+				if err != nil {
+					return err
+				}
+				err = ipTable.AppendUnique("nat", "PREROUTING", "--in-interface", forwardDevice, "-d", constants.ShootSvcNetworkMapped, "-j", "NETMAP", "--to", cfg.ShootServiceNetwork.String())
+				if err != nil {
+					return err
+				}
+				err = ipTable.AppendUnique("nat", "POSTROUTING", "--out-interface", forwardDevice, "-s", cfg.ShootServiceNetwork.String(), "-j", "NETMAP", "--to", constants.ShootSvcNetworkMapped)
+				if err != nil {
+					return err
+				}
+				err = ipTable.AppendUnique("nat", "PREROUTING", "--in-interface", forwardDevice, "-d", constants.ShootNodeNetworkMapped, "-j", "NETMAP", "--to", cfg.ShootNodeNetwork.String())
+				if err != nil {
+					return err
+				}
+				err = ipTable.AppendUnique("nat", "POSTROUTING", "--out-interface", forwardDevice, "-s", cfg.ShootNodeNetwork.String(), "-j", "NETMAP", "--to", constants.ShootNodeNetworkMapped)
+				if err != nil {
+					return err
+				}
```

Comment:
How is this supposed to work in a dual-stack cluster? `cfg.Shoot[Pod|Service|Node]Network` is assumed to be IPv4, but why should that be the case?

Reply:
The values for `cfg.Shoot[Pod|Service|Node]Network` are read from the [shoot spec here](https://github.com/gardener/gardener/pull/11582/files#diff-9509a78311746a8a2093463394e8e006042f313d898fc981ab66765b444e35d0R45), afaik these are always IPv4 even in dual-stack cases. The v6 ranges are not part of the spec and should be assigned by the cloud  provider, right? For clarity, we could rename the values to `cfg.ShootPodNetworkV4` maybe?